File: Design Patterns/src/observer/MyButton.java
package observer;

import java.util.HashSet;
import java.util.Set;

public class MyButton {

    private Set<Observer> set = new HashSet<>();

    public void attach(Observer o) {
        set.add(o);
    }

    public void push() {
        notify_button();
    }

    private void notify_button() {
        for (Observer o : set)
            o.update();
    }
}

File: Design Patterns/basic-patterns-java-kotlin/observer/MyButton.kt
package observer

import java.util.HashSet

class MyButton {
    private val set: MutableSet<Observer> = HashSet()
    fun attach(o: Observer) {
        set.add(o)
    }

    fun push() {
        notify_button()
    }

    private fun notify_button() {
        for (o in set) o.update()
    }
}